code/Controllers/CMSMain.php

```diff
@@ -744,8 +746,36 @@ public function getEditForm($id = null, $fields = null)
             $fields->push($liveLinkField = new HiddenField("AbsoluteLink", false, $record->AbsoluteLink()));
             $fields->push($liveLinkField = new HiddenField("LiveLink"));
             $fields->push($stageLinkField = new HiddenField("StageLink"));
+            $fields->push($archiveWarningMsg = new HiddenField("ArchiveWarningMessage"));
             $fields->push(new HiddenField("TreeTitle", false, $record->TreeTitle));
 
+            // Get all page's descendants
+            $record->collateDescendants(true, $descendants);
+            if (!$descendants) {
+                $descendants = [];
+            }
+
+            // Get all campaigns that the page and its descendants belong to
+            $inChangeSetIDs = ChangeSetItem::get_for_object($record)->column('ChangeSetID');
+            foreach ($descendants as $page) {
+                $inChangeSetIDs = array_merge($inChangeSetIDs, ChangeSetItem::get_for_object($page)->column('ChangeSetID'));
+            }
+            $inChangeSets = ChangeSet::get()->filter(['ID' => $inChangeSetIDs, 'State' => ChangeSet::STATE_OPEN]);
```

Comment:
Is it necessary here since the whole block will be irrelevant when the descendants is `[]`?